### ISSUE TITLE
Best practice for index pages

### ISSUE THREAD

**Issue by tm-kn** [wagtail/wagtail-personalisation#189]
I've been wondering what's the best practice for displaying variant pages that are children of an index page and are displayed on that index page. Currently the best solution in my opinion is to use `utils.exclude_variant_pages` on each queryset I want to display publicly, but it seems a bit tedious to do. Do you have any better ideas or options? Probably worth documenting as well.

**Comment by bmoe872** [comment]
I'm running into a similar requirement right now, and I'm curious on what you mean by `utils.exclude_variant_pages` or how you're using that?

Right now my index page is showing variants of child pages, and I'd like to exclude those.

**Comment by tm-kn** [comment]
Sorry, I mean this - https://github.com/wagtail/wagtail-personalisation/blob/43b5b62e6021759a77671d36f7432ab10947ebff/src/wagtail_personalisation/utils.py#L101-L113

You can use that function to filter out the QuerySet from the variant pages, just display canonical pages. Variant pages should be not visible to users as separate pages.

**Comment by bmoe872** [comment]
@tm-kn yeah I figured that out pretty quickly after I sent the first post. Sorry about that. I edited my original, because I am also curious if that is best practice.
